src/ui/src/core/index.ts

```diff
@@ -53,7 +53,7 @@ export function generateCore() {
 	 */
 	const mode: Ref<"run" | "edit"> = ref(null);
 	const writerApplication = ref<
-		{ id: string; organizationId: string } | undefined
+		{ id: string; organizationId: string } | { id: string; organizationId: string; apiKey: string } | undefined
```

Comment:
I guess it can simply be:
```suggestion
		{ id: string; organizationId: string; apiKey?: string } | undefined
```